src/System.Private.CoreLib/shared/System/IO/TextWriter.cs

```diff
@@ -283,6 +283,17 @@ public virtual void Write(object value)
             }
         }
 
+        /// <summary>
+        /// Equivalent to Write(stringBuilder.ToString()) however it uses the
+        /// StringBuilder.GetChunks() method to avoid creating the intermediate string 
+        /// </summary>
+        /// <param name="value">The string (as a StringBuilder) to write to the stream</param>
+        public virtual void Write(StringBuilder value)
+        {
+            foreach (ReadOnlyMemory<char> chunk in value.GetChunks())
```

Comment:
This will null ref if `value` is null.  Should there be argument validation?

Reply:
> which means it inevitably, (without mechanical enforcement) we will be inconsistent

I don't know, we're pretty darn consistent.  I'm sure there are places here and there where this is missed, but 99.9% of such cases throw an ArgumentNullException, plus or minus some amount in my made up percentage :)  It seems really strange to me to break from that consistency here.

I think the default should be to do such validation, and if you want to omit it, that's a discussion to have as part of the API review.